cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java

```diff
@@ -230,16 +290,49 @@ public void create(String namespace, String name, Handler<AsyncResult<Void>> han
         }
     };
 
+    private final CompositeOperation<TopicController> deleteTopicController = new CompositeOperation<TopicController>() {
+
+        @Override
+        public Future<?> composite(String namespace, ClusterOperation<TopicController> clusterOp) {
+            TopicController topicController = clusterOp.cluster();
+            Future<Void> fut = deploymentOperations.delete(namespace, topicController.getName());
+            return fut;
+        }
+
+        @Override
+        public ClusterOperation<TopicController> getCluster(String namespace, String name) {
+            Deployment dep = deploymentOperations.get(namespace, TopicController.topicControllerName(name));
+            return new ClusterOperation<>(TopicController.fromDeployment(namespace, name, dep), null);
+        }
+    };
+
     @Override
     protected void delete(String namespace, String name, Handler<AsyncResult<Void>> handler) {
-        execute(CLUSTER_TYPE_KAFKA, OP_DELETE, namespace, name, deleteKafka, ar -> {
-            if (ar.failed()) {
-                handler.handle(ar);
-            } else {
-                execute(CLUSTER_TYPE_ZOOKEEPER, OP_DELETE, namespace, name, deleteZk, handler);
-            }
-        });
 
+        Consumer<Void> execute = v -> {
```

Comment:
I don't understand why you're using a `Consumer` here. I would just factor out a `deleteKafkaAndZk()` method.

Reply:
Yes that is the other way. It's called only inside the `delete` method and not reused more times, it was my rationale. I can factor out `deleteKafkaAndZk()` but out of curiosity, do you see any drawbacks with `Consumer` ?

Reply:
Well, a private method is just clearer, it seems to me. 

Also, using lamdbas unnecessarily ends up wasting metaspace, which doesn't matter for an individual use, but if you do it lots it will add up.